edx_toggles/toggles/internal/waffle/legacy.py

```diff
@@ -100,13 +100,29 @@ def __init__(self, waffle_namespace, switch_name, module_name=None):
             waffle_namespace = waffle_namespace.name
         set_custom_attribute("deprecated_edx_toggles_waffle", "WaffleSwitch")
 
-        # Non-namespaced flag_name attribute preserved for backward compatibility
-        self.switch_name = switch_name
         name = "{}.{}".format(waffle_namespace, switch_name)
         super().__init__(name, module_name=module_name)
 
+    @property
+    def switch_name(self):
+        """
+        Non-namespaced flag_name attribute preserved for backward compatibility.
+        """
+        set_custom_attribute(
+            "deprecated_waffle_legacy_method",
+            "WaffleSwitch[{}].switch_name".format(self.name),
+        )
+        return self.name.split(".")[1]
```

Comment:
I wouldn't change behavior and chance bugs.  Are there names with multiple `.`s?

Can we just adjust above to use:
```
        self._switch_name = switch_name
```
And the return `self._switch_name`?

Reply:
Also, please add for `flag_name` as well.

Reply:
:ok_hand: